flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterStopWithSavepointIT.java

```diff
@@ -194,6 +197,30 @@ private void throwingExceptionOnCallbackWithRestartsHelper(final boolean termina
 		assertThat(getJobStatus(), either(equalTo(JobStatus.CANCELLING)).or(equalTo(JobStatus.CANCELED)));
 	}
 
+	@Test
+	public void testRestartCheckpointCoordinatorIfStopWithSavepointFails() throws Exception {
+		setUpJobGraph(ExceptionOnCallbackStreamTask.class, RestartStrategies.noRestart());
```

Comment:
I think the fact that you are using `ExceptionOnCallbackStreamTask` might be a bit misleading, as as far as I understand, you are doing it only for the `checkpointsToWaitFor` right? And you are ignoring the whole "exception on notifyCheckpointComplete callback" part of the `ExceptionOnCallbackStreamTask`?

If that's the case, I think it would be better to move/extract the count down latch to a new  
```
public static class CheckpointCountingTask extends NoOpStreamTask {
}
```
which would be just a `NoOpStreamTask` with the the `CountDownLatch`.